syntax = "proto3";

// Proto file for fileserver-client connections

package pfsfile;

service PFSFileServer {
    rpc Ping (PingRequest) returns (PingResponse) {}
    rpc Initialize (InitRequest) returns (InitResponse) {}
    rpc CreateFilePart (CreateFilePartRequest) returns (CreateFilePartResponse) {}
}

message PingRequest {
}
message PingResponse {
    string message = 1;
}

message InitRequest {
    int32 placeholder = 1;
}
message InitResponse {
    string message = 1;
}

message CreateFilePartRequest {
    string filename = 1;
    bytes data = 2;
}
message CreateFilePartResponse {
    string message = 1;
    string status_code = 2;
}